src/sql/src/statements/delete.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use sqlparser::ast::{Expr, ObjectName, Statement, TableFactor};
+use sqlparser::parser::ParserError;
+
+use crate::error::Result;
+use crate::statements::table_idents_to_full_name;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Delete {
+    // Can only be sqlparser::ast::Statement::Delete variant
+    pub inner: Statement,
+}
```

Comment:
What about holding supported fields in `Statement::Delete`?
```suggestion
pub struct Delete {
    table_name: TableFactor,
    selection: Option<Expr>,
}
```

We could also return error in `TryFrom` if the statement contains some unsupported fields, such as `using` and `returning`.

Reply:
Good idea!